draft/implementation-notes/index.html

```diff
@@ -602,7 +602,154 @@ <h2>Forward Delta</h2>
+          example, the identifier could be used as an an eTag for a REST API implementation. A 
+          good temporary directory name should be "deposit_transaction ID_OCFL Object ID". 
+          If there is guaranteed to only be a single process controlling updates to the contents 
+          of an OCFL storage root, then it is permissible to use "deposit_OCFL Object ID" as a 
+          temporary version directory name. If temporary directories are stored within OCFL 
+          Objects then the OCFL Object ID part may be omitted since it is implicit.
+          </li>
+          <li>
+          Temporary Inventory Files: A top level object inventory file should never be updated in situ.
+          Instead, the inventory should be created within the temporary version directory and updated as
+          the version contents are modified. This provides some level of forensic recovery information in 
+          the event of failure during version creation, yet ensures that the top level inventory remains 
+          valid for read clients for as long as possible. When the construction of the new version 
+          and its inventory is complete, it can then be copied to a temporary file name alongside 
+          the top level inventory. Temporary inventory files always occur in OCFL Object Roots and
+          should be names "tmp_inventory.json".
+          </li>
+        </ul>
+      </section>
 
+      <section>
+      <h2>Operational Logic</h2>
+      <p>
+      Bearing in mind the temporary names specified above, it is possible to sketch out how  
+      they can be used to ensure some level of integrity in during OCFL updates.
+      </p>
+
+        <section>  
+          <h2>Initiating a New Object Version</h2>
+          <ol>
+            <li>    
+            Generate a new transaction ID
+            </li>
+            <li>    
+            Does a temporary version directory or temporary inventory file for the object in question exist?
+	    If yes, then abort with error. Do not start a new transaction without rereading the updated object
+            </li>
+            <li>
+            Create a temporary version directory and save the transaction ID
+            </li>
+            <li>
+            Copy inventory from object into temporary directory
+            </li>
+          </ol>      
+        </section>
+
+        <section>  
+          <h2>Updating a New Object Version</h2>
+          <ol>
+            <li>    
+            Get current transaction ID
+            </li>
+            <li>
+            Does a temporary version directory for the object and transaction ID in question exist?
+	    If no, then abort with error. The transaction ID is invalid for some reason, need to debug!
+            </li>
+            <li>
+            Update object as described earlier.
+            </li>
+            <li>
+            Ideally, update the local inventory after each file change operation, not at the end
```

Comment:
Not sure whether this is ideal. If one thinks of an atomic update of many files then it is just a waste of time

Reply:
Agreed. Advantages both ways depending on use case. Scrapped as we don;t really need to say anything about it.